packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-credit-button-strategy.ts

```diff
@@ -78,6 +103,189 @@ export default class PaypalCommerceCreditButtonStrategy implements CheckoutButto
         }
     }
 
+    private async _onCancel() {
+        const state = this._store.getState();
+        const consignment = state.consignments.getConsignmentsOrThrow()[0];
+        const { shippingAddress } = consignment;
+
+        if (shippingAddress) {
+            const { email } = this._store.getState().billingAddress.getBillingAddressOrThrow();
+            const resetAddress = this._resetAddress({ ...shippingAddress, email });
+            await this._store.dispatch(this._billingAddressActionCreator.updateAddress(resetAddress));
+            await this._store.dispatch(this._consignmentActionCreator.updateAddress(resetAddress));
+
+            await this._updateOrder();
+        }
```

Comment:
The main issue that we try to cover here is to avoid passing fake data to the checkout page in cases when the user cancels PayPal Flow... And we have a case with digital Items where consignment may not have shipping address 🤔

In this case we can go in other way. We can use Billing address, because it has all related data that we need in this flow. Lets update this code with: 

```suggestion
        const state = this._store.getState();
        const { email, address } = state.billingAddress.getBillingAddressOrThrow();

        if (address) {
            const resetAddress = this._resetAddress({ ...address, email });
            
            await this._store.dispatch(this._billingAddressActionCreator.updateAddress(resetAddress));
            await this._store.dispatch(this._consignmentActionCreator.updateAddress(resetAddress));
            await this._updateOrder();
        }
```

Reply:
fixed

Reply:
fixed